### PR TITLE
Improved sync/async benchmarks

### PR DESCRIPTION
**Pull request by segabriel**

### CODE REVIEW

File: benchmarks-api/src/main/java/io/scalecube/benchmarks/BenchmarksState.java [hunk 1]
```diff
@@ -207,12 +209,18 @@ public final void runForSync(Function<SELF, Function<Long, Object>> func) {
...
+      Flux.fromStream(LongStream.range(0, settings.numOfIterations()).boxed())
+          .parallel()
+          .runOn(scheduler())
+          .map(unitOfWork)
+          .doOnTerminate(latch::countDown)
+          .subscribe();
+
+      latch.await(settings.executionTaskDuration().toMillis(), TimeUnit.MILLISECONDS);
+    } catch (InterruptedException e) {
+      e.printStackTrace();
```
Comment (artem-v) [inline benchmarks-api/src/main/java/io/scalecube/benchmarks/BenchmarksState.java]: Better just propagate with `Exceptions.propagate(e)` .   Can u put right after this comment an improvement  whihch this parallel  thing is giving to us.  I heard u told it's like 2x perf boost. Ok, can u put the stats right here. Like, here it was, here it became.

File: benchmarks-api/src/main/java/io/scalecube/benchmarks/BenchmarksState.java [hunk 2]
```diff
@@ -279,7 +293,7 @@ public final void runForAsync(Function<SELF, Function<Long, Publisher<?>>> func)
 
       BiFunction<Long, T, Publisher<?>> unitOfWork = func.apply(self);
 
-      Flux.interval(settings.rampUpInterval())
+      Flux.interval(Duration.ZERO, settings.rampUpInterval())
```
Comment (artem-v) [inline benchmarks-api/src/main/java/io/scalecube/benchmarks/BenchmarksState.java]: Cool!

File: benchmarks-api/src/main/java/io/scalecube/benchmarks/BenchmarksState.java [hunk 3]
```diff
@@ -239,12 +247,18 @@ public final void runForAsync(Function<SELF, Function<Long, Publisher<?>>> func)
```
Comment (artem-v) [inline benchmarks-api/src/main/java/io/scalecube/benchmarks/BenchmarksState.java]: As you have said runForAsync doesn't work well with this `parallel() runOn() `  trick . Plz revert .

**Comment by segabriel** [comment]
I run `SyncExampleBenchmarksRunner` on my laptop, and before these changes, I received the following results:
```
i.s.b.e.SyncExampleBenchmarksRunner-timer
             count = 57722532
         mean rate = 959152.09 calls/second
     1-minute rate = 925116.24 calls/second
     5-minute rate = 881552.39 calls/second
    15-minute rate = 870389.79 calls/second
               min = 649.00 nanoseconds
               max = 4780.00 nanoseconds
              mean = 684.39 nanoseconds
            stddev = 158.46 nanoseconds
            median = 660.00 nanoseconds
              75% <= 668.00 nanoseconds
              95% <= 816.00 nanoseconds
              98% <= 899.00 nanoseconds
              99% <= 941.00 nanoseconds
            99.9% <= 3837.00 nanoseconds
```

And after:

```
-- Timers ----------------------------------------------------------------------
i.s.b.e.SyncExampleBenchmarksRunner-timer
             count = 122485880
         mean rate = 2035577.46 calls/second
     1-minute rate = 1940461.86 calls/second
     5-minute rate = 1809094.25 calls/second
    15-minute rate = 1776252.26 calls/second
               min = 657.00 nanoseconds
               max = 8453.00 nanoseconds
              mean = 1000.55 nanoseconds
            stddev = 293.65 nanoseconds
            median = 993.00 nanoseconds
              75% <= 1020.00 nanoseconds
              95% <= 1048.00 nanoseconds
              98% <= 1061.00 nanoseconds
              99% <= 1076.00 nanoseconds
            99.9% <= 8453.00 nanoseconds
```

**Comment by artem-v** [comment]
> I run SyncExampleBenchmarksRunner on my laptop, and before these changes, I received the following results:

Have you noticed one thing: tps rate became larger 2x times, but latency is worsen. That's strange. Would you mind to run several times, say 3 and see is this pattern repeating, i.e. latency is indeed went downhill with this `parallael() `   thing.

**Pull request merged**